compiler/passes/scopeResolve.cpp

```diff
@@ -2042,6 +2114,73 @@ static void buildBreadthFirstModuleList(Vec<ModuleSymbol*>* modules,
   }
 }
 
+// Returns true if this module is capable of being used or traversed as part of
+// an access in the provided scope, false if the module is private and the
+// scope is not its direct parent or a sibling module when the parent of both
+// had a use of it.
+bool ModuleSymbol::isVisible(BaseAST* scope) {
+  if (hasFlag(FLAG_PRIVATE)) {
+    BaseAST* parentScope = getScope(defPoint);
+    INT_ASSERT(parentScope != NULL); // Should be true, given we found this
+    // module symbol.
+
```

Comment:
Perhaps moving the !hasFlag(FLAG_PRIVATE) case to be the first branch would accomplish the same purpose?